src/net_processing.cpp

```diff
@@ -939,7 +936,7 @@ void PeerLogicValidation::FinalizeNode(const CNode& node, bool& fUpdateConnectio
     if (state->fSyncStarted)
         nSyncStarted--;
 
-    if (state->nMisbehavior == 0 && state->fCurrentlyConnected /* && !node.IsBlockOnlyConn() */) {
+    if (state->nMisbehavior == 0 && node.fSuccessfullyConnected /* && !node.IsBlockOnlyConn() */) {
```

Comment:
original 20291 also changes order nMisbehaviour <> fSucessfullyConnected

Reply:
Resolved in latest push